### PR TITLE
NH Education Tax Credit

### PR DESCRIPTION
**Pull request by qingqiuzhang**
### Summary
📄🎓💰

This pull request adds the New Hampshire Education Tax Credit policy to the model. It creates a new parameter file `rate.yaml` and a new variable file `nh_education_tax_credit.py` that define and calculate the tax credit based on the state's rules.

> _`New Hampshire` code_
> _Adding tax credit variable_
> _Winter policy_

### Walkthrough
*  Implement the New Hampshire Education Tax Credit policy in the model ([link](https://github.com/PolicyEngine/policyengine-us/pull/2337/files?diff=unified&w=0#diff-9963f36048dbe7b0cf28523cc7af4714c442a41b5bc9d91ec9682b0dd7418f2fR1-R17), [link](https://github.com/PolicyEngine/policyengine-us/pull/2337/files?diff=unified&w=0#diff-7b99a6bbd7888ecf4142da1befca7016e09739aeaae23ff31dc28979592ffef0R1-R30))
  - Add a new parameter file `rate.yaml` that defines the rate of the tax credit based on the donation amount and the filing status ([link](https://github.com/PolicyEngine/policyengine-us/pull/2337/files?diff=unified&w=0#diff-9963f36048dbe7b0cf28523cc7af4714c442a41b5bc9d91ec9682b0dd7418f2fR1-R17))
  - Add a new variable file `nh_education_tax_credit.py` that calculates the tax credit amount for each tax unit using the parameter file and other variables ([link](https://github.com/PolicyEngine/policyengine-us/pull/2337/files?diff=unified&w=0#diff-7b99a6bbd7888ecf4142da1befca7016e09739aeaae23ff31dc28979592ffef0R1-R30))

#2336

### CODE REVIEW

File: policyengine_us/variables/gov/states/nh/tax/credits/nh_education_tax_credit.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from policyengine_us.model_api import *
+
+
+class nh_education_tax_credit(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "New Hampshire Education Tax Credits"
+    documentation = "New Hampshire Education Tax Credit"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nh/tax/credits/nh_education_tax_credit.py:8]:
```suggestion
    
```

File: policyengine_us/variables/gov/states/nh/tax/credits/nh_education_tax_credit.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from policyengine_us.model_api import *
+
+
+class nh_education_tax_credit(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "New Hampshire Education Tax Credits"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nh/tax/credits/nh_education_tax_credit.py:7]:
```suggestion
    label = "New Hampshire Education Tax Credit"
```

File: changelog_entry.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - NH Education Tax Credit.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - New Hampshire Education Tax Credit.
```

File: policyengine_us/parameters/gov/states/nh/tax/credits/education_tax_credit/rate.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+description: New Hampshire matches this share of Education Tax Credit, depending on the donation.
+metadata:
+  threshold_unit: currency-USD
+  rate_unit: /1
+  threshold_period: year
+  type: marginal_amount
+  label: New Hampshire Education Tax Credit 
+  reference: 
+    - title: TITLE V- TAXATION - CHAPTER 77-G - EDUCATION TAX CREDIT - Section 77-G:4
+    - href: https://www.gencourt.state.nh.us/rsa/html/V/77-G/77-G-4.htm
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nh/tax/credits/education_tax_credit/rate.yaml:10]: Also this source doesn't support the numbers here
```suggestion
      href: https://www.gencourt.state.nh.us/rsa/html/V/77-G/77-G-4.htm 
```

File: policyengine_us/parameters/gov/states/nh/tax/credits/education_tax_credit/rate.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+description: New Hampshire matches this share of Education Tax Credit, depending on the donation.
+metadata:
+  threshold_unit: currency-USD
+  rate_unit: /1
+  threshold_period: year
+  type: marginal_amount
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nh/tax/credits/education_tax_credit/rate.yaml:6]: I'm surprised `marginal_amount` worked; @nikhilwoodruff if you specify a `rate_unit: /1` does that make it a marginal rate?
```suggestion
  type: marginal_rate
```

File: policyengine_us/parameters/gov/states/nh/tax/credits/education_tax_credit/rate.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nh/tax/credits/education_tax_credit/rate.yaml:1]: rename folder to `education`

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/nh/tax/credits/education_tax_credit/rate.yaml (hunk 5 above)
Comment (nikhilwoodruff) [reply policyengine_us/parameters/gov/states/nh/tax/credits/education_tax_credit/rate.yaml:6]: No, marginal amount scales are their own scale type defined differently to marginal rate scales. The rate_unit is purely metadata and doesn't affect logic. Seems like it should be marginal rate instead?

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**